module/modmanager/manager.go

```diff
@@ -849,3 +928,16 @@ func DepsToNames(deps resource.Dependencies) []string {
 	}
 	return depStrings
 }
+
+func getModuleDataParentDirectory(options modmanageroptions.Options) string {
```

Comment:
Any commentary on what `Parent` means here? The return value on 942 includes `moduleDataFolderName` as a non-terminating directory. I would expect the phrase "module data parent directory" in this case to be `options.ViamHomeDir` as that's the parent of "module data folder name".

Reply:
Also confused about what `Parent` means here.

Reply:
I think I get the relationship now. I believe the directory layout of viam-server data is the following:
```
- viamHomeDir/ (e.g: /home/robot/.viam)
-- <stuff>
-- moduleDataParentDir/ (e.g: /home/robot/.viam/module-data -- stored on the modmanager)
--- module1Data/ (`module.dataDir` variable, passed to the module on startup)
---- <files>
--- module2Data/
---- <files>
--- module3Data/
---- <files>
```

So the "parent dir" is really the "root" directory where module data directories go. Or the parent of the yet to be evaluated module data dirs.

Except I spoke a white lie. Presumably because multiple viam-servers can run on the same machine (more likely with local developer workflows I presume), we actually have a "robot id" directory. So the path to a module is:
```
/home/robot/.viam/module-data/robot.id/module1Data/
```

I wonder if we should instead be swapping the robot id and module data:
```
/home/robot/.viam/robot.id/module-data/module1Data/
```

But definitely don't want to muck with any precedence there.

Reply:
>I wonder if we should instead be swapping the robot id and module data:
```
/home/robot/.viam/robot.id/module-data/module1Data/
```
But definitely don't want to muck with any precedence there.

That seems like a good idea at face value... but I haven't had time to think it though and that is decently far from both what was scoped and what the current pattern is in the `.viam` directory... However, open to it.


> So the "parent dir" is really the "root" directory where module data directories go. Or the parent of the yet to be evaluated module data dirs.

That is exactly correct. That is the relationship. The `moduleDataParentDirectory` is the parent to all of the module's `data_dir`s. 

Do you have a suggestion as to how I should name this to be less confusing? (because clearly it is confusing)

Reply:
Gotcha; both your comments are helpful, thanks. I was confused because the word "parent" is used to describe the module manager's relationship to modules (modules are aware of a "parent address" that is the sock file the viam server/module manager is sending gRPC requests from). `moduleDataParentDirectory` seems fitting now that you two have described the file layout. I have no strong opinion on inverting robot ID and module data for the directory structure.